src/nxdoc/nuxeo-server/installation/nuxeo-cluster-architecture/nuxeo-architecture-components/asynchronous-processing-architecture.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Asynchronous Processing
+description:
+review:
+    comment: ''
+    date: '2021-03-10'
+    status: ok
+toc: true
+tree_item_index: 700
+---
+
+Asynchronous processing relies on the **WorkManager**, **Nuxeo Stream**, **KeyValue Store** and the **PubSub Service**.
+
+## Concepts
+
+When running in cluster mode, the Nuxeo nodes need to communicate so the following services work in a distributed way:
+
+- The **[WorkManager]({{page page='work-and-workmanager'}})** can distribute its Works among nodes and share a common state.
+- **[Nuxeo Stream]({{page page='nuxeo-stream'}})** and the **[Bulk Service]({{page page='bulk-action-framework'}})** distribute processing among nodes and handle failover.
+- The **KeyValue Store** enabling distributed cache and a shared [Transient Store]({{page page='transient-store'}}).
+- The **PubSub Service** can publish messages to all nodes and is used for cache invalidation.
+
+Nuxeo Platform have built-in critical features (Document trashing, CSV export, document deletion etc.) relying on the Nuxeo Platform Bulk Service. The bulk service is implemented with Nuxeo Stream that provides two implementation:
+- Chronicle Queue
+- Kafka  
+
+<div class="table-scroll">
+<table class="hover">
+<tbody>
+<tr>
+  <td colspan="1"></th>
+  <th colspan="1">WorkManager</th>
+  <th colspan="1">Nuxeo Stream/Bulk Service</th>
+  <th colspan="1">KeyValue Store</th>
+  <th colspan="1">PubSub Service</th>
+</tr>
+<tr>
+  <th colspan="1">Redis</th>
+  <td colspan="1">Yes</td>
+  <td colspan="1">No</td>
+  <td colspan="1">Yes</td>
+  <td colspan="1">Yes</td>
+</tr>
+<tr>
+  <th colspan="1">Kafka</th>
+  <td colspan="1">Yes</td>
+  <td colspan="1">Yes</td>
+  <td colspan="1">No</td>
+  <td colspan="1">Yes</td>
+</tr>
+<tr>
+  <th colspan="1">Chronicle Queue</th>
+  <td colspan="1">No</td>
+  <td colspan="1">Yes</td>
+  <td colspan="1">No</td>
+  <td colspan="1">No</td>
+</tr>
+<tr>
+  <th colspan="1">MongoDB</th>
+  <td colspan="1">No</td>
+  <td colspan="1">No</td>
+  <td colspan="1">Yes</td>
+  <td colspan="1">No</td>
+</tr>
+<tr>
+  <th colspan="1">SQL DB</th>
+  <td colspan="1">No</td>
+  <td colspan="1">No</td>
+  <td colspan="1">Yes</td>
+  <td colspan="1">No</td>
+</tr>
+</tbody>
+</table>
+</div>
+
+## Recommendations
```

Comment:
I think I would write the production setup we recommend in this section:
- Kafka for WorkManager / BAF / PubSub
- MongoDB or SQL for KV